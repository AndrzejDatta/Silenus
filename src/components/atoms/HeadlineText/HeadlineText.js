import React from "react";
import PropTypes from "prop-types";
import { StyledHeadlineText } from "./HeadlineText.styles";
const HeadlineText = ({ text, isHeadlineTextOnBackground }) => {
  return (
    <StyledHeadlineText isHeadlineTextOnBackground={isHeadlineTextOnBackground}>
      {text}
    </StyledHeadlineText>
  );
};

HeadlineText.defaultProps = {
  text: "Headline",
  isHeadlineTextOnBackground: false,
};
HeadlineText.propTypes = {
  /** text of headline */
  text: PropTypes.string.isRequired,
  /** bool which decides about color (white (if inBackground) or green) */
  isHeadlineTextOnBackground: PropTypes.bool,
};

export default HeadlineText;
