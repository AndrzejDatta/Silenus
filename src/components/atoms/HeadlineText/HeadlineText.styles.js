import styled from "styled-components";

export const StyledHeadlineText = styled.h1`
  letter-spacing: 1px;
  font-size: ${({ theme }) => theme.font.size.large};
  color: ${(props) =>
    props.isHeadlineTextOnBackground
      ? "white"
      : ({ theme }) => theme.color.fontGreen};
  font-family: ${({ theme }) => theme.font.family.poppins};
  font-weight: normal;
`;
