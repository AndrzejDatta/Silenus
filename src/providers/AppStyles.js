import { ThemeProvider } from 'styled-components';
import { theme } from 'assets/styles/theme';
import { GlobalStyle } from 'assets/styles/GlobalStyles';
export const AppStyles = ({ children }) => {
  return (
    <ThemeProvider theme={theme}>
      <GlobalStyle />
      {children}
    </ThemeProvider>
  );
};
