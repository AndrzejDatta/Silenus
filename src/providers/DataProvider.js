import { createContext, useState } from "react";
import { dataPL } from "assets/data/data";

export const dataContext = createContext({
  startPage: {},
  loginPage: {},
  Dashboard: {},
  MojeDoniczki: {},
  DodajDoniczke: {},
  MojeRośliny: {},
  KalendarzOpieki: {},
  KontoUżytkownika: {},
  ZidentyfikujRośline: {},
  PotDetails: {},
  PlantDetails: {},
  Navbar: {},
  images: {},
  navIcons: {},
  potIcons: {},
  categoryIcons: {},
  navIconsColor: [],
  setNavIconColor: () => {},
  changeColor: () => {},
  importAll: () => {},
});

const objectName = [
  "startPage",
  "loginPage",
  "Dashboard",
  "MojeDoniczki",
  "DodajDoniczke",
  "MojeRośliny",
  "KalendarzOpieki",
  "KontoUżytkownika",
  "ZidentyfikujRośline",
  "PotDetails",
  "PlantDetails",
  "Navbar",
];

export const DataProvider = ({ children }) => {
  for (let i = 0; i < dataPL.length; i++) {
    objectName[i] = { ...dataPL[i] };
  }
  const [
    startPage,
    loginPage,
    Dashboard,
    MojeDoniczki,
    DodajDoniczke,
    MojeRośliny,
    KalendarzOpieki,
    KontoUżytkownika,
    ZidentyfikujRośline,
    PotDetails,
    PlantDetails,
    Navbar,
  ] = objectName;

  const [navIconsColor, setNavIconColor] = useState([
    {
      name: "plant",
      src: "potted_plant.svg",
      isClicked: false,
    },
    {
      name: "calendar",
      src: "calendar.svg",
      isClicked: false,
    },
    {
      name: "camera",
      src: "camera.svg",
      isClicked: false,
    },
    {
      name: "user",
      src: "accountCircle.svg",
      isClicked: false,
    },
  ]);

  function importAll(r) {
    let images = {};
    r.keys().map((item, index) => {
      images[item.replace("./", "")] = r(item);
    });
    return images;
  }

  const images = importAll(
    require.context(
      "assets/icons/userAccountOption",
      false,
      /\.(png|jpe?g|svg)$/
    )
  );
  const navIcons = importAll(
    require.context("assets/icons/navIcons", false, /\.(png|jpe?g|svg)$/)
  );

  const potIcons = importAll(
    require.context("assets/icons/potStatus", false, /\.(png|jpe?g|svg)$/)
  );

  const categoryIcons = importAll(
    require.context("assets/icons/categoryIcons", false, /\.(png|jpe?g|svg)$/)
  );

  function changeColor(nameElement) {
    setNavIconColor((prevData) =>
      prevData.map((item) => {
        if (item.name === nameElement && !item.isClicked) {
          return {
            ...item,
            isClicked: true,
          };
        } else {
          return {
            ...item,
            isClicked: false,
          };
        }
      })
    );
  }

  return (
    <dataContext.Provider
      value={{
        startPage,
        loginPage,
        Dashboard,
        DodajDoniczke,
        MojeDoniczki,
        MojeRośliny,
        KalendarzOpieki,
        KontoUżytkownika,
        ZidentyfikujRośline,
        PotDetails,
        PlantDetails,
        Navbar,
        images,
        navIcons,
        potIcons,
        categoryIcons,
        navIconsColor,
        setNavIconColor,
        changeColor,
        importAll,
      }}
    >
      {children}
    </dataContext.Provider>
  );
};

//TODO:improve way
