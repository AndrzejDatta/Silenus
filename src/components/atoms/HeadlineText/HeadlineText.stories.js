import React from 'react';
import { storiesName } from 'helpers/storiesName';
import HeadlineText from './HeadlineText';

export default {
  title: `${storiesName.atoms}HeadlineText`,
  component: HeadlineText,
  argTypes: {},
};

const Template = (args) => <HeadlineText {...args} />;

export const Default = Template.bind({});

Default.args = {};
