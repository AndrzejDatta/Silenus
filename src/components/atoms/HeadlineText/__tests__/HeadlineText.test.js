import React from "react";
import renderer from "react-test-renderer";
import "jest-styled-components";
import { AppStyles } from "providers/AppStyles";
import { render, screen } from "@testing-library/react";
import HeadlineText from "components/atoms/HeadlineText/HeadlineText";

describe("Test HeadlineText Component", () => {
  it("Test default style", () => {
    const tree = renderer
      .create(
        <AppStyles>
          <HeadlineText />
        </AppStyles>
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
    expect(tree).toHaveStyleRule("color", "#3AA688");
  });
  it("Test HeadlineText Component when text is on colorful background", () => {
    render(
      <AppStyles>
        <HeadlineText isHeadlineTextOnBackground />
      </AppStyles>
    );
    expect(screen.getByText("Headline")).toHaveStyleRule("color", "white");
  });
  it("Test setting text in HeadlineText Component", () => {
    render(
      <AppStyles>
        <HeadlineText text="Test" />
      </AppStyles>
    );
    expect(screen.getByText("Test")).toHaveStyleRule("color", "#3AA688");
  });
});
