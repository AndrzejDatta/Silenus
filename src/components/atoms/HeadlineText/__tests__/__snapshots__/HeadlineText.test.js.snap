// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test HeadlineText Component Test default style 1`] = `
.c0 {
  -webkit-letter-spacing: 1px;
  -moz-letter-spacing: 1px;
  -ms-letter-spacing: 1px;
  letter-spacing: 1px;
  font-size: 18px;
  color: #3AA688;
  font-family: "Poppins",sans-serif;
  font-weight: normal;
}

<h1
  className="c0"
>
  Headline
</h1>
`;
